src/components/date_time.rs

```diff
@@ -41,24 +40,43 @@ pub enum CalendarDateTime {
     /// Conversion from [`chrono::DateTime<Utc>`](DateTime) results in this variant. Use
     /// `date_time.with_timezone(&Utc)` to convert `date_time` from arbitrary time zone to UTC.
     Utc(DateTime<Utc>),
+    /// `FORM #3: DATE WITH LOCAL TIME AND TIME ZONE REFERENCE`: refers to a time zone definition.
+    WithTimezone {
+        /// The date and time in the given time zone.
+        date_time: NaiveDateTime,
+        /// The ID of the time zone definition in a VTIMEZONE calendar component.
+        tzid: String,
+    },
 }
 
 impl CalendarDateTime {
-    pub(crate) fn from_str(s: &str) -> Option<Self> {
-        if let Ok(naive_date_time) = NaiveDateTime::parse_from_str(s, NAIVE_DATE_TIME_FORMAT) {
+    pub(crate) fn from_property(property: &Property) -> Option<Self> {
+        let value = property.value();
+        if let Some(tzid) = property.params().get("TZID") {
+            Some(Self::WithTimezone {
+                date_time: NaiveDateTime::parse_from_str(value, NAIVE_DATE_TIME_FORMAT).ok()?,
+                tzid: tzid.value().to_owned(),
+            })
+        } else if let Ok(naive_date_time) =
+            NaiveDateTime::parse_from_str(value, NAIVE_DATE_TIME_FORMAT)
+        {
```

Comment:
`else if let` ?
sure this code doesn't want to be a `match statement?
...checking...
pfff, lets just leave this here for now

Reply:
I'm not sure how you'd write this as a match statement, I'm trying two different things.